src/bootstrap/src/core/build_steps/check.rs

```diff
@@ -381,7 +381,7 @@ impl Step for RustAnalyzer {
         let mut cargo = prepare_tool_cargo(
             builder,
             compiler,
-            Mode::ToolStd,
+            Mode::ToolRustc,
```

Comment:
Couple lines above there is `builder.ensure(Std::new(target));`. Could you please replace it with `builder.ensure(Rustc::new(target, builder));`?